src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/pages/remoteapi/RemoteAPIDetailsPage.java

```diff
@@ -56,76 +83,75 @@ public void confirmDelete() {
 		confirmButton.click();
 	}
 
-	public ApiStatus getRemoteApiStatus() {
-		WebElement connectionStatus = (new WebDriverWait(driver, 10)).until(ExpectedConditions
-				.presenceOfElementLocated(By.className("status-label")));
-
-		String labelClass = connectionStatus.getAttribute("class");
-		if (labelClass.contains("api-connected")) {
-			return ApiStatus.CONNECTED;
-		} else if (labelClass.contains("api-invalid")) {
-			return ApiStatus.INVALID;
-		} else if (labelClass.contains("api-error")) {
-			return ApiStatus.ERROR;
+	public boolean isRemoteAPIConnected() {
+		try {
+			return remoteStatusConnected.isDisplayed();
+		} catch (NoSuchElementException e) {
+			return false;
 		}
-
-		throw new ElementNotVisibleException("Coudldn't get api status");
 	}
 
 	public void clickConnect() {
-		WebElement connectButton = (new WebDriverWait(driver, 10)).until(ExpectedConditions.elementToBeClickable(By
-				.className("oauth-connect-link")));
-		connectButton.click();
-
-		waitForAjax();
+		remoteStatusConnect.click();
 
 	}
 
 	public void clickAuthorize() {
-		driver.switchTo().frame("oauth-connect-frame");
+		String parentWindowHandler = driver.getWindowHandle(); // Store your parent window
+		String subWindowHandler = null;
+
+		Set<String> handles = driver.getWindowHandles(); // get all window handles
+		for (String handle : handles) {
+			subWindowHandler = handle;
+		}
+		driver.switchTo().window(subWindowHandler); // switch to popup window
+
+		// Now you are in the popup window, perform necessary actions here
+
 		WebElement authorizeButton = driver.findElement(By.id("authorize-btn"));
 		authorizeButton.click();
 
-		driver.switchTo().defaultContent();
+
+		driver.switchTo().window(parentWindowHandler);  // switch back to parent window
 
 		waitForTime(8000);
 	}
 
-	public boolean checkDeleteSuccess() {
-		boolean deleted = false;
-
-		logger.debug("Checking for client existence");
-		do {
-			try {
-				WebElement el = driver.findElement(By.className("ant-page-header-heading-title"));
-				if (el.getText().equals("Remote IRIDA Connections")) {
-					logger.debug("Successfully loaded client list page");
-					waitForAjax();
-					logger.debug("Table loaded");
-					List<WebElement> findElements = driver.findElements(By.className("remoteApiCol"));
-					deleted = true;
-					// check if the element is in the table
-					for (WebElement ele : findElements) {
-						if (ele.getText().equals(clientId)) {
-							deleted = false;
-						}
-					}
-				}
-			} catch (StaleElementReferenceException e) {
-				logger.debug("Got stale element reference exception when trying to get text on h1, trying again.");
-			}
-		} while (!deleted);
-
-
-		return deleted;
-	}
+//	public boolean checkDeleteSuccess() {
```

Comment:
Remove commented out code or re-enable if needed please.

Reply:
Removed in e965419